### PR TITLE
Add: anyhost key judgment

### PR DESCRIPTION
**Pull request by willson-chen**
**What this PR does**:

Add anyhost judgment in ToConfigurators in registry/base_configuration_listener.go, which is used to improve code quality by removing 'TODO' comment

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by willson-chen** [comment]
I removed the `TODO` comment and add the anyhost judgment, which follows the [java implement ](https://github.com/apache/dubbo/blob/b8b42b16182b8fde245dd6c5edeb1115955d1004/dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/Configurator.java#L86).

File: registry/base_configuration_listener.go [hunk 1]
```diff
@@ -100,7 +100,13 @@ func ToConfigurators(urls []*common.URL, f func(url *common.URL) config_center.C
 			configurators = []config_center.Configurator{}
 			break
 		}
-		//TODO:anyhost_key judage
+
+		var override map[string][]string = url.GetParams()
```
Comment (AlexStocks) [inline registry/base_configuration_listener.go:104]: override := url.GetParams()

**Comment by willson-chen** [comment]
> Based on https://github.com/apache/dubbo-go/blob/master/contributing.md to change the title

Thanks. I will pay attention to the rule in next PR.

File: registry/base_configuration_listener.go (hunk 1 above)
Comment (willson-chen) [reply registry/base_configuration_listener.go]: Done. Thanks. :)

File: registry/base_configuration_listener.go [hunk 2]
```diff
@@ -100,7 +100,13 @@ func ToConfigurators(urls []*common.URL, f func(url *common.URL) config_center.C
 			configurators = []config_center.Configurator{}
 			break
 		}
-		//TODO:anyhost_key judage
+
+		override := url.GetParams()
+		delete(override, constant.ANYHOST_KEY)
+		if len(override) == 0 {
+			configurators = nil
```
Comment (zouyx) [inline registry/base_configuration_listener.go:107]: It seems will clear `configurators` when url.GetParams()  is nil. Is it right logic?
Like:

Fisrt loop: url.GetParams() is not null, add a element to `configurators`.
Second loop :url.GetParams() is null ,will clear `configurators`?

Comment (willson-chen) [reply registry/base_configuration_listener.go:107]: I am also puzzled by the java implememt that I follow. Could some one help to check it?

Comment (chickenlj) [reply registry/base_configuration_listener.go:107]: > It seems will clear configurators when url.GetParams() is nil. Is it right logic?

I think `configurators  = nil` should be removed.

Comment (zouyx) [reply registry/base_configuration_listener.go:107]: > > It seems will clear configurators when url.GetParams() is nil. Is it right logic?
> 
> I think `configurators = nil` should be removed.

Wilson will fix it.

**Comment by willson-chen** [comment]
Sorry for my lateness. Change request done.

**Pull request merged**